libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -560,28 +579,76 @@ impl NodeAPI for Greenlight {
         Ok(())
     }
 
-    async fn sweep(
-        &self,
-        to_address: String,
-        fee_rate_sats_per_vbyte: u64,
-    ) -> Result<WithdrawResponse> {
+    async fn sweep(&self, req: SweepRequest) -> Result<WithdrawResponse> {
         let mut client = self.get_client().await?;
 
         let request = pb::WithdrawRequest {
             feerate: Some(pb::Feerate {
-                value: Some(pb::feerate::Value::Perkw(fee_rate_sats_per_vbyte * 250)),
+                value: Some(pb::feerate::Value::Perkw(req.sats_per_vbyte * 250)),
             }),
             amount: Some(Amount {
                 unit: Some(Unit::All(true)),
             }),
-            destination: to_address,
+            destination: req.to_address,
             minconf: None,
             utxos: vec![],
         };
 
         Ok(client.withdraw(request).await?.into_inner())
     }
 
+    async fn prepare_sweep(&self, req: PrepareSweepRequest) -> Result<PrepareSweepResponse> {
+        let funds = self.list_funds().await?;
+        let utxos = self.utxos(funds).await?;
+
+        let mut amount: u64 = 0;
+        let txins: Vec<TxIn> = utxos
+            .iter()
+            .map(|utxo| {
+                amount += utxo.amount_millisatoshi;
+                TxIn {
+                    previous_output: OutPoint {
+                        txid: Txid::from_slice(&utxo.txid).unwrap(),
+                        vout: 0,
+                    },
+                    script_sig: Script::new(),
+                    sequence: Sequence(0),
+                    witness: Witness::default(),
+                }
+            })
+            .collect();
+
+        // remove millisats lower than 1 satoshi (1-999 msat)
+        amount /= 1000;
+        amount *= 1000;
+
+        let btc_address = Address::from_str(&req.to_address)?;
+        let tx_out: Vec<TxOut> = vec![TxOut {
+            value: amount,
+            script_pubkey: btc_address.payload.script_pubkey(),
+        }];
+        let mut tx = Transaction {
+            version: 2,
+            lock_time: bitcoin::PackedLockTime(0),
+            input: txins.clone(),
+            output: tx_out,
+        };
+
+        let witness_input_size: u64 = 110;
+        let tx_weight = tx.strippedsize() as u64 * WITNESS_SCALE_FACTOR as u64
+            + witness_input_size * txins.len() as u64;
+        let fee: u64 = tx_weight * req.sats_per_vbyte / WITNESS_SCALE_FACTOR as u64;
+        if fee >= amount {
```

Comment:
This is not enough to validate the transaction. I see there is a `verify` function that can be used.

Reply:
Can you point out where is it @roeierez ?
I'm mimicking swap and it is not calling verify https://github.com/breez/breez-sdk/blob/4fdeb41997df2953a04efb48520447e4e3f9cd46/libs/sdk-core/src/swap.rs#L606